### PR TITLE
chore: bump to use go1.23

### PR DESCRIPTION
**Pull request by chenrui333**
- bump to use go1.23
- also update setup-go to use go-version-file to simplify go version management

relates to https://github.com/Homebrew/homebrew-core/pull/183138

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -34,12 +34,12 @@ jobs:
...
           - amd64
           - arm64
-    
+
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
-          go-version: '1.21'
+          go-version-file: go.mod
```
Comment (Duologic) [inline .github/workflows/docker.yml:42]: Perhaps also incldue cache-dependency-path:

```suggestion
          go-version-file: go.mod
          cache-dependency-path: go.sum
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
-          go-version: '1.21'
+          go-version-file: go.mod
```
Comment (Duologic) [inline .github/workflows/release.yml:20]: And here:

```suggestion
          go-version-file: go.mod
          cache-dependency-path: go.sum
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
-          go-version: '1.21'
+          go-version-file: go.mod
```
Comment (Duologic) [inline .github/workflows/tests.yml:13]: And here:

```suggestion
          go-version-file: go.mod
          cache-dependency-path: go.sum
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ jobs:
...
```
Comment (Duologic) [inline .github/workflows/tests.yml:22]: And here:

```suggestion
          go-version-file: go.mod
          cache-dependency-path: go.sum
```

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -28,6 +28,6 @@ jobs:
...
```
Comment (Duologic) [inline .github/workflows/tests.yml:31]: And here:

```suggestion
          go-version-file: go.mod
          cache-dependency-path: go.sum
```

**Review by Duologic — APPROVED** [review]
Looks alright, left a few suggestions.

**Comment by malcolmholmes** [comment]
Thanks for the contribution. Closed via #485.

**Pull request closed without merging**